### DISCUSSION TITLE
Adding API support for user related operations in libreswan

### DISCUSSION THREAD

**Discussion by swiq-anas-abdullah** [libreswan/libreswan#1198] [Polls]
I was thinking about using libreswan for whitelabel purpose. I found out that currently libreswan does't provide REST API support for people to use it with there own solutions. The API would let backend applications to manage user related operations on libreswan. Some of the use cases could be

- Managing Users
- Managing Users certificates
- Other Admin related APIs for use with admin dashboards

You can comment any use case that you may have in mind

**Comment by letoams** [comment]
On Mon, 17 Jul 2023, swiq-anas-abdullah wrote:

> Subject: [libreswan/libreswan] Adding API support for user related operations

> I was thinking about using libreswan for whitelabel purpose. I found out that currently libreswan does't provide API support for people to use it with
> there own solutions.

What do you mean with API? Do you mean an HTTP API?

The internal API for libreswan is "ipsec whack". But it is not a public
API and is not guaranteed to not change.

We were planning to add an API (not HTTP API) using varlink.org, but it
seems varlink failed to replace dbus/systemd, and the existing APIs
using dbus/systemd are all terrible, usually assume Freedesktop API (eg
graphics) and we haven't found something better yet.

Paul

**Reply by swiq-anas-abdullah** [reply]
Hey Paul
I mean HTTP REST API , which can used by backend applications to manage the vpn services. I will edit the title

**Comment by paulwouters** [comment]
There is a very old project that we are hoping to revive but for which we have not had the resources to do so:

https://github.com/Rishabh04-02/Libreswan-managing-interface/blob/master/INSTALLATION_INSTRUCTIONS.md

You are welcome to work on that and modernize it :)

I believe cockpit also has some libreswan support buildin ?

**Reply by swiq-anas-abdullah** [reply]
What do think regarding creating a seperate project for it with new tech stack (node.js and react js). We can create REST APIs which can be used by clients and we can create a modern UI with React JS (or Nest JS). As there are alot of javascript devs out there , it will be easy to manage the project in future also. This would be a more modularized solution

**Reply by paulwouters** [reply]
what would the backend be? node is just not secure enough to run as backend

**Reply by swiq-anas-abdullah** [reply]
golang or python can also be used on backend.
